Mathlib/Data/Set/Image.lean

```diff
@@ -518,6 +518,16 @@ theorem image_preimage_eq {f : α → β} (s : Set β) (h : Surjective f) : f ''
     ⟨y, (e.symm ▸ hx : f y ∈ s), e⟩
 #align set.image_preimage_eq Set.image_preimage_eq
 
+@[simp]
+theorem preimage_subset_preimage {f : α → β} (s t : Set β) (h : Surjective f) :
```

Comment:
What if you make `preimage_subset_preimage_iff` simp instead of adding a new lemma?

Reply:
That requires another `@[simp]` lemma to prove the precondition of `preimage_subset_preimage_iff`. Is it advisable to have `simp` be responsible for a chain of reasoning `f.surjective` -> `s ⊆range f` -> `f ⁻¹' s ⊆ f ⁻¹' t ↔ s ⊆ t` like this? I don't know how brittle simp's precondition search is

```lean
@[simp]
theorem Set.subset_range_of_surjective {f : α → β} (h : Surjective f) (s : Set β) :
    s ⊆ range f := Surjective.range_eq h ▸ subset_univ s
```

[Playground](https://live.lean-lang.org/#code=import%20Mathlib.Data.Set.Image%0Aimport%20Mathlib.Tactic%0A%0Aopen%20Set%20Function%0A%0Aattribute%20%5Bsimp%5D%20preimage_subset_preimage_iff%0A%0A%40%5Bsimp%5D%0Atheorem%20Set.subset_range_of_surjective%20%7Bf%20%3A%20%CE%B1%20%E2%86%92%20%CE%B2%7D%20(h%20%3A%20Surjective%20f)%20(s%20%3A%20Set%20%CE%B2)%20%3A%0A%20%20%20%20s%20%E2%8A%86%20range%20f%20%3A%3D%20Surjective.range_eq%20h%20%E2%96%B8%20subset_univ%20s%0A%0Aexample%20%7Bf%20%3A%20%CE%B1%20%E2%86%92%20%CE%B2%7D%20(s%20t%20%3A%20Set%20%CE%B2)%20(h%20%3A%20Surjective%20f)%20%3A%0A%20%20%20%20f%20''%20(f%20%E2%81%BB%C2%B9'%20s)%20%E2%8A%86%20t%20%E2%86%94%20f%20''%20(f%20%E2%81%BB%C2%B9'%20s)%20%E2%8A%86%20t%20%3A%3D%20by%0A%20%20conv%20%3D%3E%0A%20%20%20%20congr%0A%20%20%20%20%C2%B7%20simp%20%5Bh%2C%20-image_preimage_eq%5D%0A%20%20%20%20%C2%B7%20simp%20%5Bh%2C%20-image_subset_iff%5D%0A)

Reply:
I think this lemma is worth having because `s ⊆ range f` preconditions are pretty common.